cmd/dlt/machinepool/cmd.go

```diff
@@ -27,43 +28,69 @@ import (
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var Cmd = &cobra.Command{
-	Use:     "machinepool ID",
-	Aliases: []string{"machinepools", "machine-pool", "machine-pools"},
-	Short:   "Delete machine pool",
-	Long:    "Delete the additional machine pool from a cluster.",
-	Example: `  # Delete machine pool with ID mp-1 from a cluster named 'mycluster'
-  rosa delete machinepool --cluster=mycluster mp-1`,
-	Run: run,
-	Args: func(_ *cobra.Command, argv []string) error {
-		if len(argv) != 1 {
-			return fmt.Errorf(
-				"Expected exactly one command line parameter containing the id of the machine pool",
-			)
-		}
-		return nil
-	},
-}
+const (
+	use     = "machinepool ID"
+	short   = "Delete machine pool"
+	long    = "Delete the additional machine pool from a cluster."
+	example = `  # Delete machine pool with ID mp-1 from a cluster named 'mycluster'
+  rosa delete machinepool --cluster=mycluster mp-1`
+)
+
+var (
+	aliases = []string{"machinepools", "machine-pool", "machine-pools"}
+)
+
+func NewDeleteMachinePoolCommand() *cobra.Command {
+	options := NewDeleteMachinepoolUserOptions()
+	var cmd = &cobra.Command{
+		Use:     use,
+		Aliases: aliases,
+		Short:   short,
+		Long:    long,
+		Example: example,
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), DeleteMachinePoolRunner(options)),
+		Args:    cobra.MaximumNArgs(1),
+	}
+	flags := cmd.Flags()
+	flags.StringVar(
+		&options.machinepool,
+		"machinepool",
+		"",
+		"Machine pool of the cluster to target",
+	)
 
-func init() {
-	ocm.AddClusterFlag(Cmd)
-	confirm.AddFlag(Cmd.Flags())
+	ocm.AddClusterFlag(cmd)
+	confirm.AddFlag(cmd.Flags())
+	return cmd
 }
 
-func run(_ *cobra.Command, argv []string) {
-	r := rosa.NewRuntime().WithAWS().WithOCM()
-	defer r.Cleanup()
+func DeleteMachinePoolRunner(userOptions DeleteMachinepoolUserOptions) rosa.CommandRunner {
+	return func(_ context.Context, runtime *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		options := NewDescribeMachinepoolOptions()
+		// Allow the use also directly the machine pool id as positional parameter
+		if len(argv) == 1 && !cmd.Flag("machinepool").Changed {
+			userOptions.machinepool = argv[0]
+		} else {
+			err := cmd.ParseFlags(argv)
```

Comment:
@hunterkepley This is as a result of not taking in the pointer to the options. If you take in the `pointer`, you can remove the additional `cmd.ParseFlags` call and the additional assignment to the `userOptions.machinepool` field. This will be handled for you by cobra.